stock_component/models/stock_move.py

```diff
@@ -25,3 +31,33 @@ def _has_components(self):
 
     def generate_component_moves(self):
         self.mapped("move_line_ids.lot_id").pull_components()
+
+    def _is_pulling_components(self):
+        return self._context.get(PULLING_COMPONENTS, False)
+
+    def _check_no_component_moved(self):
+        serials = self.mapped("move_line_ids.lot_id")
+        component_serials = serials.filtered(lambda s: s.is_component)
+
+        if component_serials:
+            message = _(
+                "You are attempting to move products that are components "
+                "of an equipment."
+            )
+            details = "\n".join(
+                [
+                    self._get_component_serial_not_movable_message(s)
+                    for s in component_serials
+                ]
+            )
+            raise ValidationError("\n".join((message, details)))
+
+    def _get_component_serial_not_movable_message(self, serial):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5280787)